### ISSUE TITLE
question about using detach()

### ISSUE THREAD

**Issue by btwbtm** [researchmm/TTSR#28]
Hi, I am confused.
Why do you use detach here? Sicne you have disable gradient of self.vgg19, it shouldn't make any difference if you remove detach.
https://github.com/researchmm/TTSR/blob/a3c618d011ef40b0f83004bf9bdbd545e1735ca7/trainer.py#L91
And here, you do not use detach. What is the difference?
https://github.com/researchmm/TTSR/blob/a3c618d011ef40b0f83004bf9bdbd545e1735ca7/trainer.py#L89
Thx!

**Comment by FuzhiYang** [comment]
setting requires_grad = False for VGG19 only freezes the parameters of VGG19. However, the gradients can still be backpropagated to the proper layers in the SR network.

**Comment by btwbtm** [comment]
How about line 91, if we remove detach(), what's the difference?

**Comment by FuzhiYang** [comment]
In this code, removing detach() of hr makes no difference since hr itself is a leaf node tensor
